src/lib/plot.typ

```diff
@@ -27,102 +27,147 @@
+/// #show-parameter-block("minor-tick-step", ("none", "auto", "float"), default: "none", [
+///   Like `tick-step`, but for minor tick marks.])
+/// #show-parameter-block("ticks", ("none", "array"), default: "none", [
+///   List o custom tick marks in addition to those generated by `tick-step` and `minor-tick-step`.
+///   This argument supports an array of `float` on where to place marks, or an
+///   array of `(<float>, <content>)` tuples, for setting custom tick mark labels per mark.
+///   Examples: `(1, 2, 3)` or `((1, [One]), (2, [Two]), (3, [Three]))`])
+/// #show-parameter-block("format", ("none", "string", "function"), default: "float", [
+///   Specifies the tick label formatting or a custom formatter function. The following
+///   predefined formats are supported:
+///     / `"float"`: Floating point formatting rounded to two digits after the point (see `decimals`)
+///     / `"sci"`: Scientific formatting with $times 10^n$ used as expoent syntax
+///     / `number => content`: A function that takes a number and returns content that gets uesd
+///       as tick label
+/// ])
+/// #show-parameter-block("decimals", ("int"), default: "2", [
+///   Number of decimals digits to display for tick labels, if the format is set
+///   to `"float"`.
+/// ])
+/// #show-parameter-block("unit", ("none", "content"), default: "none", [
+///   Suffix to append to all tick labels.
+/// ])
+/// #show-parameter-block("grid", ("bool", "string"), default: "false", [
+///   If `true` or `"major"`, show grid lines for all major ticks. If set
+///   to `"minor"`, show grid lines for minor ticks only.
+///   The value `"both"` enables grid lines for both, major- and minor ticks.
+/// ])
 ///
-/// - body (body): Calls of `plot.add` or `plot.add-*` commands
-/// - size (array): Plot canvas size tuple of width and height in canvas units
+/// - body (body): Calls of `plot.add` or `plot.add-*` commands. Note that normal drawing
+///   commands like `line` or `rect` are not allowed insides the plots body, instead wrap
+///   them in `plot.add-annotation`, which lets you select the axes used for drawing.
+/// - size (array): Plot size tuple of `(<width>, <height>)` in canvas units.
+///   This is the plots inner plotting size without axes and labels.
 /// - axis-style (none, string): Axis style "scientific", "left", "school-book"
-///     - `"scientific"`: Frame plot area and draw axes y, x, y2, and x2 around it
-///     - `"school-book"`: Draw axes x and y as arrows with both crossing at $(0, 0)$
-///     - `"left"`: Draw axes x and y as arrows, the y axis stays on the left (at `x.min`)
+///     / `"scientific"`: Frame plot area using a rect and draw axes `x` (bottom), `y` (left), `x2` (top), and `y2` (right) around it.
+///       If `x2` or `y2` are unset, they mirror their opposing axis.
+///     / `"scientific-auto"`: Draw set (used) axes `x` (bottom), `y` (left), `x2` (top) and `y2` (right) around
```

Comment:
Scientific shows all four axes (x, y, x2, y2), while scientific-auto only shows those in use. E.G. if you use x, y and y2, U-shaped axes are drawn instead of a rectangle.

Reply:
(That style is in PR #337, not yet available on this branch.)